File: Practice/atcoder/ABC/027/src/b.py
# 全体の人数を島の数で割り切れる場合のみ可能
# 少なくても多くても隣の島に橋をかけなくてはならないのでそうする
# いやよく考えたら累積しながらで判断できるか

n = int(input())
arr = list(map(int, input().split()))
if sum(arr) % n != 0:
    print(-1)
    exit()
if sum(arr) == 0:
    print(0)
    exit()

divider = sum(arr) // n
ans = 0
cur = 0
for i in range(n):
    cur += arr[i]
    if cur != (i + 1) * divider:
        ans += 1

print(ans)

File: Practice/atcoder/ABC/027/src/b.go
package main

import "fmt"

func main() {
	var N int
	fmt.Scan(&N)
	A, S := make([]int, N), 0
	for i := 0; i < N; i++ {
		var V int
		fmt.Scan(&V)
		A[i] = V
		S += V
	}

	if S%N != 0 {
		fmt.Println(-1)
		return
	}
	D := S / N
	ANS := 0
	L, T := 1, 0
	for _, a := range A {
		T += a
		// if T/L != D {
		if T != D*L {
			ANS += 1
		}
		L += 1

	}
	fmt.Println(ANS)
}
